src/scraper/scraper.cpp

```diff
@@ -2776,7 +2778,23 @@ bool LoadProjectObjectToStatsByCPID(const std::string& project, const CSerialize
 bool ProcessProjectStatsFromStreamByCPID(const std::string& project, boostio::filtering_istream& sUncompressedIn,
                                          const double& projectmag, const BeaconMap& mBeaconMap, ScraperStats& mScraperStats)
 {
-    std::vector<std::string> vXML;
+
+    // We need a temporary scraper stats map to hold the CPID entries and compute the statistics
+    // because if the project is greylisted, we will only commit some of these to the main scraper
+    // statistics map.
+    ScraperStats mScraperStatsTemp;
+
+    // Get the whitelist and greylist... this is primarily to determine greylisted projects to eliminate
+    // the CPID level statistics.
+
+    const NN::WhitelistSnapshot projectWhitelist = NN::GetWhitelist().Snapshot();
+
+    const NN::GreylistSnapshot greylist = NN::Quorum::FilterGreylist(projectWhitelist);
+
+    // If the specified project is not on the whitelist, immmediately return false.
+    if (!projectWhitelist.Contains(project)) return false;
+
+    bool greylisted = greylist.Contains(project);
```

Comment:
`GreylistSnapshot::Contains()` just forwards to `WhitelistSnapshot::Contains()` for code that just needs to check whether the project existed when the snapshot was taken. I should have made the name more clear. To check greylist status, use `GreylistSnapshot::CheckActive()` or `GreylistSnapshot::CheckGreylisted()`.

Here, we want to use: 
```c++
bool greylisted = greylist.CheckGreylisted(project);
```

Reply:
You're right. I even recoded the filter when I changed the reasons to a vector. I must just be tired. A general state these days unfortunately.